src/main/java/com/hubspot/smtp/client/SmtpSession.java

```diff
@@ -53,6 +59,13 @@
       SmtpCommand.QUIT,
       SmtpCommand.NOOP);
 
+  private static final Supplier<Executor> SHARED_DEFAULT_EXECUTOR = Suppliers.memoize(SmtpSession::getSharedExecutor);
+
+  private static ExecutorService getSharedExecutor() {
+    ThreadFactory threadFactory = new ThreadFactoryBuilder().setDaemon(true).setNameFormat("niosmtpclient-%d").build();
+    return Executors.newCachedThreadPool(threadFactory);
```

Comment:
the only nit here is that I would set allow core timeout to prevent threads from being un-killlable. E.g.:

```java

ExecutorService executorService = Executors.newCachedTheradPool(threadFactory);
if (executorService instanceof ThreadPoolExecutor) {
  ((ThreadPoolExecutor)executor).allowCoreThreadTimeout(true);
} else {
  throw new IllegalArgumentException("Expected Executors.newCachedThreadPool to return a ThreadPoolExecutor, but it returned " + String.valueOf(executorService));
}
```